libopenage/util/misc_test.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+#include "misc.h"
+
+#include "../testing/testing.h"
+
+
+namespace openage {
+namespace util {
+namespace tests {
+
+
+void array_conversion() {
+	{
+		static const std::vector<uint8_t> test_array8{
+			0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef,
+			0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7
+		};
+		// The same array but as uint64_t
+		// ASDF: Can we just assume that everything is little-endian?
+		static const std::vector<uint64_t> test_array64{
+			0xefcdab8967452301, 0xf7f6f5f4f3f2f1f0
+		};
+
+		array8_to_array64(test_array8.data(), 16) == test_array64 or TESTFAIL;
+		array64_to_array8(test_array64.data(), 2) == test_array8 or TESTFAIL;
+	}
+	{
+		static const std::vector<uint8_t> test_array8{
+			0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef,
+			0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0x00, 0x00
+		};
+		// The same array but as uint64_t
+		// ASDF: Can we just assume that everything is little-endian?
+		static const std::vector<uint64_t> test_array64{
+			0xefcdab8967452301, 0x0000f5f4f3f2f1f0
+		};
+
+		array64_to_array8(test_array64.data(), 2) == test_array8 or TESTFAIL;
+		// Now I'm converting 14 instead of 16 bytes, the remaining ones should be
+		// filled with zeroes
+		array8_to_array64(test_array8.data(), 14) == test_array64 or TESTFAIL;
+	}
+	{
+		static const std::vector<uint8_t> test_array8{
+			0x01, 0x23,
+			0x45, 0x67, 0x89, 0xab, 0xcd, 0xef, 0xf0, 0xf1,
+			0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7
+		};
+		// The same array but as uint64_t
+		// ASDF: Can we just assume that everything is little-endian?
+		static const std::vector<uint64_t> expected{
+			0xf1f0efcdab896745, 0x0000f7f6f5f4f3f2
+		};
+
+		// Now I'm trying to convert misaligned bytes
+		// ASDF: Or It's just nonsense?
+		array8_to_array64(test_array8.data() + 2, 14) == expected or TESTFAIL;
```

Comment:
I think you are right. I've been reading a little about that and now I understand better. Anyway now I'm not using memcpy (and reinterpreted types) anymore because of endianness.